source/client/sni_utility.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "nighthawk/common/uri.h"
+
+#include "external/envoy/include/envoy/http/protocol.h"
+
+namespace Nighthawk {
+namespace Client {
+
+class SniUtility : public Envoy::Logger::Loggable<Envoy::Logger::Id::main> {
```

Comment:
Curious, looks like we ended up defining a class for this anyways. I guess it is for the purposes of the logging mix-in. Is creation of a class the only way to use the Envoy logging facility? Or is there some other shape or form of it that we could use for plain utility functions that don't require to be on a class?

Reply:
That is a good question. I think the mix-in is the way it is supposed to be used.
We could create one or more classes that mix these in just dedicated to logging?
e.g.
```c++
class MainLog : public Envoy::Logger::Loggable<Envoy::Logger::Id::main> {
public:
 // bunch of helpers to log at various log levels, possibly shared from a base class
}
```

Reply:
As discussed, this is something we can explore in the future.